src/Integration/Suppression/SuppressedIssuesFileStorage.cs

```diff
@@ -0,0 +1,113 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    internal class SuppressedIssuesFileStorage : ISuppressedIssuesFileStorage
+    {
+        private readonly string fileDirectory;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        public SuppressedIssuesFileStorage(ILogger logger) : this(logger, new FileSystem())
+        {
+            
+        }
+
+        internal SuppressedIssuesFileStorage(ILogger logger, IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+
+            fileDirectory = Path.Combine(Path.GetTempPath(), "SLVS", "Roslyn");
+        }
+
+        public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
+        {
+            try
+            {
+                if (!ValidateFileName(sonarProjectKey))
+                {
+                    return Enumerable.Empty<SonarQubeIssue>();
+                }
+
+                var filePath = GetFilePath(sonarProjectKey);
+                var fileContent = fileSystem.File.ReadAllText(filePath);
+                return JsonConvert.DeserializeObject<IEnumerable<SonarQubeIssue>>(fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"SuppressedIssuesFileStorage Get Error: {ex.Message}");
+            }
+
+            return Enumerable.Empty<SonarQubeIssue>();
+        }
+
+        public void Update(string sonarProjectKey, IEnumerable<SonarQubeIssue> allSuppressedIssues)
+        {
+            try
+            {
+                if (!ValidateFileName(sonarProjectKey))
+                {
+                    return;
+                }
+
+                var filePath = GetFilePath(sonarProjectKey);
+                var fileContent = JsonConvert.SerializeObject(allSuppressedIssues);
+                fileSystem.Directory.CreateDirectory(fileDirectory);
+                fileSystem.File.WriteAllText(filePath, fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"SuppressedIssuesFileStorage Update Error: {ex.Message}");
+            }
+        }
+
+        private bool ValidateFileName(string sonarProjectKey)
+        {
+            if (string.IsNullOrWhiteSpace(sonarProjectKey))
+            {
+                logger.WriteLine("SuppressedIssuesFileStorage: File name should not be empty");
+                return false;
+            }
+            if (sonarProjectKey.IndexOfAny(Path.GetInvalidFileNameChars()) >= 0)
```

Comment:
This is an interesting case. Could we support keys with invalid characters by replacing the invalid characters with valid ones?

Reply:
We can replace them but could it cause key duplication?

Reply:
It could but it's pretty unlikely to happen in practice, and it is still better than failing to suppress if the key contains a single invalid character.

There are other places we use the project key in a file name. I didn't think we handled invalid characters properly, but it looks like we do - there is a `PathHelper.EscapeFileName` method in `Core`.